src/ansys/fluent/core/utils/file_transfer_service.py

```diff
@@ -15,8 +22,239 @@ def __init__(self):
         super().__init__("PyPIM is not configured.")
 
 
+class FiletransferStrategy(Protocol):
+    """Provides the file transfer strategy."""
+
+    def upload(
+        self, file_name: Union[list[str], str], remote_file_name: Optional[str] = None
+    ) -> None:
+        """Upload file to the server."""
+        ...
+
+    def download(
+        self, file_name: Union[list[str], str], local_directory: Optional[str] = None
+    ) -> None:
+        """Download file from the server."""
+        ...
+
+
+class LocalFileTransferStrategy(FiletransferStrategy):
+    """Provides the local file transfer strategy."""
+
+    def __init__(self):
+        self.fluent_cwd = (
+            pathlib.Path(str(get_fluent_exe_path()).split("fluent")[0]) / "fluent"
+        )
+
+    def file_exists_on_remote(self, file_name: str) -> bool:
+        """Check if remote file exists.
+
+        Parameters
+        ----------
+        file_name: str
+            File name.
+
+        Returns
+        -------
+            Whether file exists.
+        """
+        full_file_name = pathlib.Path(self.fluent_cwd) / os.path.basename(file_name)
+        return full_file_name.is_file()
+
+    def upload(
+        self, file_name: Union[list[str], str], remote_file_name: Optional[str] = None
+    ) -> None:
+        local_file_name = pathlib.Path(file_name)
+        if local_file_name.exists() and local_file_name.is_file():
+            if remote_file_name:
+                shutil.copyfile(file_name, str(self.fluent_cwd / f"{remote_file_name}"))
+            else:
+                shutil.copyfile(file_name, str(self.fluent_cwd / f"{file_name}"))
+
+    def download(
+        self, file_name: Union[list[str], str], local_directory: Optional[str] = None
+    ) -> None:
+        dest_path = pathlib.Path(os.getcwd()) / f"{file_name}"
+        local_file_name = (
+            pathlib.Path(local_directory) if local_directory else dest_path
+        )
+        if local_file_name.exists() and local_file_name.samefile(file_name):
+            return
+        shutil.copyfile(file_name, str(local_file_name))
+
+
+def _get_files(file_name: Any):
```

Comment:
We are now able to specify a more specific list of possible types for `file_name` here.

Reply:
Yes, done. Thank you.